src/libertem/job/masks.py

```diff
@@ -78,18 +104,28 @@ def validate_mask_functions(self):
             fns = [fns]
         for fn in fns:
             try:
-                if 'self' in fn.__code__.co_freevars:
-                    log.warning('mask factory closes over self, may be inefficient')
+                # functools.partial doesn't have a __code__ attribute
+                # use fn.func.c__code__
+                # FIXME check the args and kwargs of fn for inefficiencies?
```

Comment:
My main motivation here was to avoid an AttributeError when accessing __code__ for the common case of a functools.partial function. How should I resolve this in this particular case?

Reply:
Sorry, the comment was mostly aimed at the FIXME, I think your solution for `partial` is fine.

Reply:
Resolved with size check of cloudpickle result.